### PR TITLE
Update docstrings for LTI class constructors

### PR DESCRIPTION
**Pull request by murrayrm**
This PR addresses the documentation issues raised in PR #163 related to how the `StateSpace`, `TransferFunction`, and `FRD` constructors are documented.  In this PR, the first line of the docstring is modified for the each of the above classes, as well as their `init` methods and the corresponding `ss`, `tf`, and `frd` functions to have the most common call signature.

This PR leaves in place the copy constructor functionality and only modifies the documentation.

While I was at it, I also fixed some indentation problems and other issues that were causing warnings in sphinx.

### CODE REVIEW

**Comment by murrayrm** [comment]
Added more information to main documentation on class constructors + updated docstrings for other functions with variable arguments.  The justification for including them in this PR is that they are consistent with the initial changes and the comments in PR #163.

File: doc/conventions.rst [hunk 1]
```diff
@@ -9,6 +9,97 @@ Library conventions
...
+function.  A full list of functions can be found in :ref:`function-ref`.
+
+Transfer functions
+------------------
+The :class:`TransferFunction` class is used to represent input/output
+transfer functions
+
+.. math::
+
+  G(s) = \frac{\text{num}(s)}{\text{den(s)}}
```
Comment (roryyorke) [inline doc/conventions.rst]: in first case (s) is outside `\text{}`, in second case inside it.

File: doc/conventions.rst [hunk 2]
```diff
@@ -9,6 +9,97 @@ Library conventions
 The python-control library uses a set of standard conventions for the way
 that different types of standard information used by the library.
 
+LTI system representation
+=========================
+
+Linear time invariant (LTI) systems are represented in python-control in
+state space, transfer function, or frequency response data (FRD) form.  Most
+functions in the toolbox will operate on any of these data types and
+functions for convering between between compatible types is provided.
```
Comment (roryyorke) [inline doc/conventions.rst]: converting

File: control/margins.py [hunk 3]
```diff
@@ -338,7 +336,9 @@ def phase_crossover_frequencies(sys):
 
 
 def margin(*args):
-    """Calculate gain and phase margins and associated crossover frequencies
+    """margin(sys)
```
Comment (roryyorke) [inline control/margins.py]: Just below (line 345) the name "sysdata" is used; perhaps that should be used here?

File: control/ctrlutil.py [hunk 4]
```diff
@@ -45,6 +45,11 @@
 import numpy as np
 from numpy import pi
 
+# Hack for sphinx.ext.autodoc: if numpy is a mock import, then numpy.pi 
+# will be assigned to _Mock() and this generates a type error
+if not isinstance(pi, float):
+    pi = 3.14
```
Comment (roryyorke) [inline control/ctrlutil.py]: how about "from math import pi" here?  Or does that result in a mocked instance of some sort too?

File: doc/conventions.rst [hunk 5]
```diff
@@ -9,6 +9,97 @@ Library conventions
...
+The :class:`TransferFunction` class is used to represent input/output
+transfer functions
+
+.. math::
+
+  G(s) = \frac{\text{num}(s)}{\text{den(s)}}
+       = \frac{a_0 s^n + a_1 s^{n-1} + \cdots + a_n}
+              {b_0 s^m + b_1 s^{m-1} + \cdots + b_m},
+
+where n is generally greater than m (for a proper transfer function).
```
Comment (roryyorke) [inline doc/conventions.rst]: maybe greater than or equal to?

File: control/ctrlutil.py (hunk 4 above)
Comment (murrayrm) [reply control/ctrlutil.py]: Much better fix since math is included in the base python distribution.

**Comment by murrayrm** [comment]
Good feedback @roryyorke.  Small updates pushed to address these.

**Pull request merged**